src/data/repositories/ValidationRuleGroupD2Repository.ts

```diff
@@ -0,0 +1,53 @@
+import { Id } from "$/domain/entities/Ref";
+import { ValidationRuleGroup } from "$/domain/entities/ValidationRuleGroup";
+import { Future } from "$/domain/entities/generic/Future";
+import { ValidationRuleRepository } from "$/domain/repositories/ValidationRuleGroupRepository";
+import { D2Api } from "$/types/d2-api";
+import { FutureData, apiToFuture } from "../api-futures";
+
+export class ValidationRuleD2Repository implements ValidationRuleRepository {
+    constructor(private api: D2Api) {}
+
+    get(): FutureData<ValidationRuleGroup[]> {
+        return apiToFuture(
+            this.api.models.validationRuleGroups
+                .get({
+                    fields: {
+                        id: true,
+                        name: true,
+                        description: true,
+                    },
+                })
+                .map(d2Response => {
+                    return d2Response.data.objects.map(ValidationRuleGroupD2Repository => {
```

Comment:
Since the `ValidationRuleGroup` entity match the values returned by the api we can directly return the whole array.

```ts
        return apiToFuture(
            this.api.models.validationRuleGroups
                .get({
                    fields: { id: true, name: true, description: true },
                })
                .map(d2Response => {
                    return d2Response.data.objects;
                })
        );
```